src/DevChatter.Bot.Web/wwwroot/js/sprite.js

```diff
@@ -1,24 +1,119 @@
-function Sprite(imageSrc, x, y, direction) {
+function Sprite(imageSrc, x, y, direction, mod) {
   this.image = new Image();
   this.image.src = imageSrc;
   this.x = x || Math.floor((Math.random() * 200) + (1920 / 2 - 100));
   this.y = y || Math.floor((Math.random() * 200) + (1080 / 2 - 100));
   this.direction = direction || this.getRandomDirection();
+  this.mod = mod || this.getRandomModifier(); //sets rate of displacement (how fast the sprite is moving)
   this.timesRendered = 0;
+  this.bounced = false;
```

Comment:
What's the `bounced` do for us here? Are we sometimes out of bounds and and on our way back in?

Reply:
It doesn't do much here, think I was just initializing the value at construction. The sprite is keeping track of when it goes passed a boundary/ hits a wall. Then it changes direction and we wait till it is back in the boundary before we let it change direction again. I actually haven't tried it without, I assumed it would get stuck calling the bounce functions if there was a flag

Reply:
It actually seems to work fine without the flag as far as I can tell. It can be taken out if you feel it's unnecessary. I'm going to go ahead add your simplified angle code to the solution.